update simspec tutorial to use quickspectra

sbailey opened the pull request:
Updates the simulating-desi-spectra tutorial to use quickspectra, while keeping the original tutorial as quickgen.ipynb.  The new tutorial:

  * generates 100 QSO spectra
  * simulates those with with dark conditions using quickspectra
  * runs redrock
  * re-simulates with very bright conditions
  * re-runs redrock
  * compares redshift performance with and without moon (hint: moon is bad, though maybe not as bad as I would have expected).

External vetting welcome, or otherwise I will self-merge by Monday and we can have fun with lots of external vetting during the tutorial session.

Full disclosure: my git-fu failed me and I ended up re-adding the original tutorial under a new name instead of moving it in a way that would preserve history.  For jupyter notebooks I don't think that is a big deal, but if someone feels strongly we could fix this properly in a different branch.

sbailey committed:
update simulating spectra tutorial to use quickspectra

sbailey committed:
re-add quickgen tutorial

moustakas commented:
This is very nice, @sbailey.

dkirkby commented:
I agree this looks great!  Some nitpicking suggestions in case you have time:
 - "read out here" -> "read about here"
 - (17.12?) -> (17.12 which will be available sometime in Dec 2017)
 - `%pylab inline` -> `%matplotlib inline` (since you already imported np and plt)
 - plot histograms of the sample redshifts and r-mag?
 - add a line showing flux=0 in `plotspec()`
 - will the srun command in cell [10] propagate the necessary DESI env?
 - your moon-up geometry is non-Euclidean!  moonsep=10 deg would fix this.
 - if you want to see a bigger moon effect, increase the observing airmass.
 - mention the mini tutorial in the appendix?

sbailey committed:
implement PR suggestions

sbailey commented:
Thanks for the suggestions; I implemented those in the notebook.

The srun command does propagate the notebook environment into the batch job, but it is possible that the user has commands in their .bashrc.ext, .bash_profile.ext, or .tcshrc.ext that could override that.  I haven't mapped out the full phase space of what works and what doesn't, so for now I just added a note about what to look for if that command fails.  I have tested this notebook with "clean" .bash*.ext files that do not configure any DESI code or environment variables by default (@tskisner finally convinced me that that really was the best thing...)
